src/graphql/data/nft/Collection.ts

```diff
@@ -0,0 +1,130 @@
+import { Trait } from 'nft/hooks/useCollectionFilters'
+import { GenieCollection } from 'nft/types'
+import { useLazyLoadQuery } from 'react-relay'
+
+import { CollectionQuery } from './__generated__/CollectionQuery.graphql'
+
+const collectionQuery = graphql`
+  query CollectionQuery($address: String!) {
+    nftCollections(filter: { addresses: [$address] }) {
+      edges {
+        cursor
+        node {
+          bannerImage {
+            url
+          }
+          collectionId
+          description
+          discordUrl
+          homepageUrl
+          image {
+            url
+          }
+          instagramName
+          isVerified
+          name
+          numAssets
+          twitterName
+          nftContracts {
+            address
+            chain
+            name
+            standard
+            symbol
+            totalSupply
+          }
+          traits {
+            name
+            values
+            stats {
+              name
+              value
+              assets
+              listings
+            }
+          }
+          markets(currencies: ETH) {
+            floorPrice {
+              currency
+              value
+            }
+            owners
+            totalVolume {
+              value
+              currency
+            }
+            listings {
+              value
+            }
+            volume(duration: DAY) {
+              value
+              currency
+            }
+            volumePercentChange(duration: DAY) {
+              value
+              currency
+            }
+            floorPricePercentChange(duration: DAY) {
+              value
+              currency
+            }
+          }
+        }
+      }
+      pageInfo {
+        endCursor
+        hasNextPage
+        hasPreviousPage
+        startCursor
+      }
+    }
+  }
+`
+
+export function useCollectionQuery(address: string): GenieCollection | undefined {
+  const queryData = useLazyLoadQuery<CollectionQuery>(collectionQuery, { address })
+
+  const queryCollection = queryData.nftCollections?.edges[0]?.node
+  const traits = {} as Record<string, Trait[]>
+  if (queryCollection?.traits) {
+    queryCollection?.traits.forEach((trait) => {
+      trait.values?.map((value) => {
+        return {
+          trait_type: trait.name,
+          trait_value: value,
+        } as Trait
+      })
+    })
+  }
+  return {
+    address,
+    isVerified: queryCollection?.isVerified ?? undefined,
+    name: queryCollection?.name ?? undefined,
+    description: queryCollection?.description ?? undefined,
+    standard: queryCollection?.nftContracts ? queryCollection?.nftContracts[0]?.standard ?? undefined : undefined,
+    bannerImageUrl: queryCollection?.bannerImage?.url,
+    stats: queryCollection?.markets
+      ? {
+          num_owners: queryCollection?.markets[0]?.owners ?? undefined,
+          floor_price: queryCollection?.markets[0]?.floorPrice?.value ?? undefined,
```

Comment:
nit: do the `queryCollection?.markets[0]` in one place and assign it to a shorter var like `market` so we don't have so much repetition here